### PR TITLE
fix Chart resource version

### LINKED ISSUE #3604: update chart version
## Bug Report

**What version of Kubernetes are you using?**
none

**What version of TiDB Operator are you using?**
v1.1.7

**What storage classes exist in the Kubernetes cluster and what are used for PD/TiKV pods?**

**What's the status of the TiDB cluster pods?**

**What did you do?**
helm version with `version.BuildInfo{Version:"v3.4.1", GitCommit:"c4e74854886b2efe3321e185578e6db9be0a6e29", GitTreeState:"clean", GoVersion:"go1.14.11"}`
**What did you expect to see?**
image

**What did you see instead?**
image

** What did I do to solve**
refer [code](https://github.com/swartz-k/tidb-operator/commit/837847c11a88fcd5aa93fa731c20312fb76376cd)

may use `for FILE in ./charts/*; do helm lint $FILE; done` to check chart error
image

### PR DESCRIPTION
**Pull request by swartz-k**
### What problem does this PR solve?
Close #3604

### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [x] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

**Comment by DanielZhangQD** [comment]
@swartz-k Thanks a lot for the contribution!
Could you please update the test result in the PR description to cover the following cases:

- Fresh installation
- Install with v1.1.8 and then upgrade with the chart in this PR

Thanks!

File: charts/tidb-operator/templates/controller-manager-rbac.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@ metadata:
     helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+"  "_" }}
 ---
 kind: ClusterRole
-apiVersion: rbac.authorization.k8s.io/v1beta1
+apiVersion: v1
```
Comment (DanielZhangQD) [inline charts/tidb-operator/templates/controller-manager-rbac.yaml:17]: It should be `rbac.authorization.k8s.io/v1`?

Comment (swartz-k) [reply charts/tidb-operator/templates/controller-manager-rbac.yaml:17]: `v1` is ok，i will add test result right now.

**Comment by DanielZhangQD** [comment]
@swartz-k Have you updated the code and run some tests?

**Comment by swartz-k** [comment]
> @swartz-k Have you updated the code and run some tests?

so sorry，had a bussines trip. Fixing

**Comment by swartz-k** [comment]
> @swartz-k Have you updated the code and run some tests?

Hey，sorry to bother you. Can you do me a favor，I dont know what happend，and jenkins-ci pending for a little long time. Thanks very much

**Comment by DanielZhangQD** [comment]
> > @swartz-k Have you updated the code and run some tests?
> 
> Hey，sorry to bother you. Can you do me a favor，I dont know what happend，and jenkins-ci pending for a little long time. Thanks very much

Done.

**Comment by DanielZhangQD** [comment]
> > @swartz-k Have you updated the code and run some tests?
> 
> so sorry，had a bussines trip. Fixing

It's OK, thanks!

**Comment by DanielZhangQD** [comment]
> @swartz-k Thanks a lot for the contribution!
> Could you please update the test result in the PR description to cover the following cases:
> 
> * Fresh installation
> * Install with v1.1.8 and then upgrade with the chart in this PR
> 
> Thanks!

@swartz-k Did you run some basic tests?

**Comment by swartz-k** [comment]
> > @swartz-k Thanks a lot for the contribution!
> > Could you please update the test result in the PR description to cover the following cases:
> > 
> > * Fresh installation
> > * Install with v1.1.8 and then upgrade with the chart in this PR
> > 
> > Thanks!
> 
> @swartz-k Did you run some basic tests?

yeah，I used this edited chart to install tidb-operator，and run tidb client to test；and also `make test`

**Comment by swartz-k** [comment]
> > > @swartz-k Thanks a lot for the contribution!
> > > Could you please update the test result in the PR description to cover the following cases:
> > > 
> > > * Fresh installation
> > > * Install with v1.1.8 and then upgrade with the chart in this PR
> > > 
> > > Thanks!
> > 
> > 
> > @swartz-k Did you run some basic tests?
> 
> yeah，I used this edited chart to install tidb-operator，and run tidb client to test；and also `make test`
> But，I found something error in tidb controller manager，error like below
> image

image: pingcap/tidb-operator:v1.1.9
```
 # /usr/local/bin/tidb-controller-manager -version
TiDB Operator Version: version.Info{GitVersion:"v1.1.9", GitCommit:"c48d54188fc19251d65e716b880e9e151bf07874", GitTreeState:"clean", BuildDate:"2020-12-28T11:50:42Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by swartz-k** [comment]
> > @swartz-k Thanks a lot for the contribution!
> > Could you please update the test result in the PR description to cover the following cases:
> > 
> > * Fresh installation
> > * Install with v1.1.8 and then upgrade with the chart in this PR
> > 
> > Thanks!
> 
> @swartz-k Did you run some basic tests?

If possible，can i run jenkins-ci test in local by using shell command or something？Those steps cost a long time,sorry

**Comment by DanielZhangQD** [comment]
> > yeah，I used this edited chart to install tidb-operator，and run tidb client to test；and also `make test`
> > But，I found something error in tidb controller manager，error like below
> > image
> 
> image: pingcap/tidb-operator:v1.1.9
> 
> ```
>  # /usr/local/bin/tidb-controller-manager -version
> TiDB Operator Version: version.Info{GitVersion:"v1.1.9", GitCommit:"c48d54188fc19251d65e716b880e9e151bf07874", GitTreeState:"clean", BuildDate:"2020-12-28T11:50:42Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
> ```

The dm parameters are added in the master branch, you can just edit the deployment of `tidb-controller-manager` to remove them.

**Comment by DanielZhangQD** [comment]
> > > @swartz-k Thanks a lot for the contribution!
> > > Could you please update the test result in the PR description to cover the following cases:
> > > 
> > > * Fresh installation
> > > * Install with v1.1.8 and then upgrade with the chart in this PR
> > > 
> > > Thanks!
> > 
> > 
> > @swartz-k Did you run some basic tests?
> 
> If possible，can i run jenkins-ci test in local by using shell command or something？Those steps cost a long time,sorry

Run jenkins-ci locally will take a long time, you can just run the two cases I have mentioned which only take a very short time:
> > > * Fresh installation
> > > * Install with v1.1.8 and then upgrade with the chart in this PR

**Comment by swartz-k** [comment]
> * Fresh installation
> * Install with v1.1.8 and then upgrade with the chart in this PR


@DanielZhangQD Hey，I did run those tests，with screenshot like below
image

**Comment by swartz-k** [comment]
> The dm parameters are added in the master branch, you can just edit the deployment of `tidb-controller-manager` to remove them.

Pardon，can I think imgae `pingcap/tidb-operator:v1.1.9` does not support param  `-dm-master-failover-period`
I pull this iamge and run command ` /usr/local/bin/tidb-controller-manager -dm-master-failover-period=5m` will fail.
image

**Comment by DanielZhangQD** [comment]
> > The dm parameters are added in the master branch, you can just edit the deployment of `tidb-controller-manager` to remove them.
> 
> Pardon，can I think imgae `pingcap/tidb-operator:v1.1.9` does not support param `-dm-master-failover-period`
> I pull this iamge and run command ` /usr/local/bin/tidb-controller-manager -dm-master-failover-period=5m` will fail.

No, it does not support, only the master branch supports this parameter.

**Comment by swartz-k** [comment]
> > > The dm parameters are added in the master branch, you can just edit the deployment of `tidb-controller-manager` to remove them.
> > 
> > 
> > Pardon，can I think imgae `pingcap/tidb-operator:v1.1.9` does not support param `-dm-master-failover-period`
> > I pull this iamge and run command ` /usr/local/bin/tidb-controller-manager -dm-master-failover-period=5m` will fail.
> 
> No, it does not support, only the master branch supports this parameter.
In this way，before new image released，people who pull the master branch，and install by default，helm install will fail

**Comment by DanielZhangQD** [comment]
> > > Pardon，can I think imgae `pingcap/tidb-operator:v1.1.9` does not support param `-dm-master-failover-period`
> > > I pull this iamge and run command ` /usr/local/bin/tidb-controller-manager -dm-master-failover-period=5m` will fail.
> > 
> > 
> > No, it does not support, only the master branch supports this parameter.
> > In this way，before new image released，people who pull the master branch，and install by default，helm install will fail

OK, we will try to improve that later, thanks!

**Review by handlerww — APPROVED** [review]
LGTM

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Pull request merged**